src/soroban/parsers.ts

```diff
@@ -139,7 +139,18 @@ function parseSuccessful(
             : xdr.ScVal.scvVoid()
         };
       })[0]
-    })
+    }),
+
+    stateChanges: sim.stateChanges?.length ?? 0 > 0
+        ? sim.stateChanges!.map((entryChange) => {
+          return {
+            type: entryChange.type,
+            key: xdr.LedgerKey.fromXDR(entryChange.key, 'base64'),
+            before: !!entryChange.before ? xdr.LedgerEntry.fromXDR(entryChange.before, 'base64') : undefined,
+            after: !!entryChange.after ? xdr.LedgerEntry.fromXDR(entryChange.after, 'base64') : undefined,
```

Comment:
if you want `before` and `after` to be omitted when they aren't present you'll need to use the spread operator (check in this file for examples, I think it's something like

```typescript
key: xdr.LedgerKey.fromXDR(entryChange.key, 'base64'),
...(entryChange.before && { before: xdr.LedgerEntry.fromXDR(entryChange.before, 'base64') },
```

and you want this for `stateChanges` too. Alternatively you can use `delete` (much easier to read lol).

Reply:
I think now there is no need to omit as these are required fields. so spread operator wont be needed here

Reply:
What I mean is that maybe we _want_ to omit them at the SDK level since the API choice of making them `null` is odd (imo).

Reply:
@Shaptic  hmmm I dont understand, what we have in response is required fields,  I will have to revert it back to previous [interface](https://github.com/stellar/js-stellar-sdk/pull/963#discussion_r1602236321) in order to omit right? 

```
before: xdr.LedgerEntry | null;
after: xdr.LedgerEntry | null;
```

I can def omit the entire stateChanges tho as it is not required field. 



> Alternatively you can use delete (much easier to read lol). 

can you give an example?